lib/Driver/UnixToolChains.cpp

```diff
@@ -212,7 +212,9 @@ toolchains::GenericUnix::constructInvocation(const DynamicLinkJobAction &job,
   }
 
   if (getTriple().getOS() == llvm::Triple::Linux &&
-      job.getKind() == LinkKind::Executable) {
+      job.getKind() == LinkKind::Executable &&
+      !context.Args.hasFlag(options::OPT_static_executable,
+                            options::OPT_no_static_executable, false)) {
     Arguments.push_back("-pie");
```

Comment:
Not really your fault, but this seems that it is an ELF specific flag, Android and Linux both support this, as do other platforms like BSDs and Haiku IIRC.  I'm not sure that there is any value to the OS check.

Reply:
Ok, I'll fix that.